src/epiviz.gl/mouse-reader.js

```diff
@@ -129,27 +130,21 @@ class MouseReader {
               .concat(
                 this._calculateViewportSpot(...getLayerXandYFromEvent(event))
               );
-            this.element.parentElement.dispatchEvent(
-              new CustomEvent("onSelection", {
-                detail: {
-                  bounds: this._currentSelectionPoints,
-                  type: this.tool,
-                },
-              })
-            );
+            this.handler.dispatchEvent.call(this.handler, "onSelection", {
```

Comment:
what is `this.handler` supposed to be? we usually dispatch an event from `this` context?

Reply:
this.handler is webgl-vis, class instance. I was mistaken I thought dispatch event is outside. I don't need to use .call and pass this context. I'll call it directly here and it would work.

Reply:
Updated